src/providers/sources/rezka.ts

```diff
@@ -0,0 +1,207 @@
+import { flags } from '@/main/targets';
+import { makeSourcerer } from '@/providers/base';
+import { EmbedScrapeContext } from '@/utils/context';
+import { NotFoundError } from '@/utils/errors';
+
+const rezkaBase = 'https://rezka.ag';
+
+const TRASH_LIST = [
+  '//_//QEBAQEAhIyMhXl5e',
+  '//_//Xl5eIUAjIyEhIyM=',
+  '//_//JCQhIUAkJEBeIUAjJCRA',
+  '//_//IyMjI14hISMjIUBA',
+  '//_//JCQjISFAIyFAIyM=',
+];
+
+const clearTrash = (data: string): string => {
+  const trashPattern = new RegExp(TRASH_LIST.join('|'), 'g');
+  const cleanedData = data.replace(trashPattern, '').replace('#h', '');
+  return atob(cleanedData) || '';
+};
+
+interface VideoQuality {
+  type: string;
+  url: string;
+}
+
+function parseVideoLinks(inputString: string): Record<number, VideoQuality> {
+  const linksArray = inputString.split(',');
+  const result: Record<number, VideoQuality> = {};
+
+  linksArray.forEach((link) => {
+    const match = link.match(/\[(\d+p)[^\]]*](https?:\/\/[^\s,]+)/);
+    if (match) {
+      const [, quality, m3u8Url] = match;
+      const numericQuality = parseInt(quality, 10);
+      result[numericQuality] = { type: 'mp4', url: m3u8Url.replace(/:hls:manifest\.m3u8$/, '') };
+    }
+  });
+
+  return result;
+}
+
+interface MovieData {
+  url: string;
+  title: string;
+  id: string | null;
+  year: number | null;
+  titleAndYear: string;
+  type: 'show' | 'movie';
+}
+
+const extractTitleAndYear = (input: string) => {
+  const regex = /^(.*?),.*?(\d{4})/;
+  const match = input.match(regex);
+
+  if (match) {
+    const [, title, year] = match;
+    return { title: title.trim(), year: year ? parseInt(year, 10) : null };
+  }
+  return null;
+};
+
+async function getMovieReleaseDatesAndUrls({
+  type,
+  title,
+  releaseYear,
+  ctx,
+}: {
+  type: 'show' | 'movie';
+  title: string;
+  releaseYear: number;
+  ctx: EmbedScrapeContext;
+}): Promise<MovieData | null> {
+  const searchUrl = `${rezkaBase}/engine/ajax/search.php?q=${encodeURIComponent(title)}`;
+
+  try {
+    const searchData = await ctx.proxiedFetcher<string>(searchUrl, { method: 'GET' });
+    const itemRegex = /<a href="([^"]+)"><span class="enty">([^<]+)<\/span> \(([^)]+)\)/g;
+    let match = itemRegex.exec(searchData);
+    const movieData: MovieData[] = [];
+
+    while (match !== null) {
+      const [, url, , titleAndYear] = match;

```

Comment:
Perhaps this would be cleaner if you did the following instead? It doesn't bother me for the other ones, but this one is a bit more complex than those.

```suggestion
      const url = match[1];
      const titleAndYear = match[3];
```